src/core/symbology/qgssymbollayerutils.h

```diff
@@ -897,6 +898,39 @@ class CORE_EXPORT QgsSymbolLayerUtils
      */
     static QgsStringMap evaluatePropertiesMap( const QMap<QString, QgsProperty> &propertiesMap, const QgsExpressionContext &context );
 
+    /**
+     * QgsSymbolLayerReference uses QgsSymbolLayer unique uuid identifier since QGIS 3.30, instead of the symbol
+     * key (rule for QgsRuleBasedRenderer for instance) and index path, so this method migrates \a mapLayers old references
+     * to new ones.
+     * \since QGIS 3.30
+     * \deprecated since QGIS 3.30
+     */
+    Q_DECL_DEPRECATED static void fixOldSymbolLayerReferences( const QMap<QString, QgsMapLayer *> &mapLayers );
```

Comment:
Maybe QgsProjectFileTransform is a better place for this function to live?

Reply:
Yes :+1: